### PR TITLE
Disable warnings when parsing possibly non-XML strings

### LINKED ISSUE #88: Warnings when parsing non-XML strings in XmlSerializer
When parsing non-XML strings in `XmlSerializer` a warning is being returned.

Example warning:

```
Message: SimpleXMLElement::__construct(): {"odata.error":{"code":"ResourceNotFound","message":{"lang":"en-US","value":"The
```
The warning is generated in `ServiceException::parseErrorMessage()` because that's possibly parsing a non-XML string (as the comment *try to parse using xml serializer, if failed, return the whole body* suggests)

### Sample backtrace:
File: /media/snafu-htdocs/report-uri/report-uri/application/libraries/Microsoft/vendor/microsoft/azure-storage/src/Common/Internal/Serialization/XmlSerializer.php
Line: 241
Function: __construct

File: /media/snafu-htdocs/report-uri/report-uri/application/libraries/Microsoft/vendor/microsoft/azure-storage/src/Common/Exceptions/ServiceException.php
Line: 89
Function: unserialize

File: /media/snafu-htdocs/report-uri/report-uri/application/libraries/Microsoft/vendor/microsoft/azure-storage/src/Common/Exceptions/ServiceException.php
Line: 70
Function: parseErrorMessage

Non XML strings should not generate a warning and I have a PR incoming where the warning is silenced by calling `libxml_use_internal_errors(true)`.

Thanks for looking into this.

### PR DESCRIPTION
**Pull request by spaze**
Fixes #88

### CODE REVIEW

**Comment by katmsft** [comment]
@spaze Please re-send the PR to dev branch.

**Comment by spaze** [comment]
@katmsft Sorry, re-based to dev now.

File: src/Common/Exceptions/ServiceException.php [hunk 1]
```diff
@@ -86,7 +86,12 @@ protected static function parseErrorMessage(ResponseInterface $response)
         $serializer = new XMLSerializer();
         $errorMessage = '';
         try {
+            $internalErrors = libxml_use_internal_errors(true);
             $parsedArray = $serializer->unserialize($response->getBody());
+            if (libxml_get_errors()) {
```
Comment (katmsft) [inline src/Common/Exceptions/ServiceException.php]: Better use this form in if statement: `if(!empty(libxml_get_errors))`.

File: src/Common/Exceptions/ServiceException.php [hunk 2]
```diff
@@ -86,7 +86,12 @@ protected static function parseErrorMessage(ResponseInterface $response)
...
+                throw new \Exception('Cannot parse XML');
```
Comment (katmsft) [inline src/Common/Exceptions/ServiceException.php]: Minor: The error message can be saved in `Resources` for easier maintenance.

Comment (katmsft) [inline src/Common/Exceptions/ServiceException.php]: Alternatively, you can construct the error message got from `libxml_get_errors()` and pass it out so user could get a clearer message.

File: src/Common/Exceptions/ServiceException.php (hunk 1 above)
Comment (spaze) [inline src/Common/Exceptions/ServiceException.php]: Resolved, used `if (!empty(libxml_get_errors()))`

File: src/Common/Exceptions/ServiceException.php (hunk 2 above)
Comment (spaze) [inline src/Common/Exceptions/ServiceException.php]: The message is in `Resources` now, and contains errors coming from `libxml_get_errors()`. The exception is caught few lines below and the message is discarded, but I agree it's better to have a meaningful message in the exception.

**Comment by spaze** [comment]
Thanks @katmsft, comments resolved, test case added, commit amended. Thanks!

**Comment by katmsft** [comment]
LGTM

**Pull request merged**